### DISCUSSION TITLE
Cannot find module 'sharp' when hosting Astro with SSR

### DISCUSSION THREAD

**Discussion by reallukedev** [vercel/vercel#8696] [Help] [answered]
👋 Hello!

Whenever I build to Vercel, using [Astro](https://astro.build/) and following [this guide](https://docs.astro.build/en/guides/deploy/vercel/), it gives me this error ![image](https://user-images.githubusercontent.com/58954397/194675658-8210c5e9-2a2c-49f6-9c07-19487dbf1fd4.png)

This is what my package.json looks like (it contains the sharp package):
![image](https://user-images.githubusercontent.com/58954397/194675675-220c3db4-e0f3-47c6-8801-483f10aa44cc.png)

This is very annoying, as I cannot access my [site](https://realluke.vercel.app/)

Thanks
 - Luke 🙂

**Comment by amyegan** [comment, marked as answer]
Hi @Reallukeisbest! Are you using yarn? There have been some cases where conflicts between yarn and some packages have caused similar dependency problems. If using yarn, do you get a different result if you change your [install command](https://vercel.com/changelog/customizing-the-install-command-while-creating-projects) to use `npm` or `pnpm`?

**Reply by reallukedev** [reply]
I am using npm

**Reply by amyegan** [reply]
It looks like your site is working now. Is that an old deployment or a workaround?

If you can share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) (or your repo if it is public), I would be happy to take a closer look at what is causing this error.

**Reply by reallukedev** [reply]
> It looks like your site is working now. Is that an old deployment or a workaround?
> 
> If you can share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) (or your repo if it is public), I would be happy to take a closer look at what is causing this error.

I just reverted the update that caused it. If you would like, I could upload the site that is causing the error on github.

**Reply by reallukedev** [reply]
> It looks like your site is working now. Is that an old deployment or a workaround?
> 
> If you can share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) (or your repo if it is public), I would be happy to take a closer look at what is causing this error.

👋 Hello, @amyegan!

[Here](https://github.com/Reallukeisbest/portfolio-error) is a repo of the website that is causing the problem. I have also just updated [my website](https://realluke.vercel.app/) to contain the breaking code.

The only lines that I added were these lines in my astro.config
```js
import { defineConfig } from 'astro/config';

import vercel from "@astrojs/vercel/serverless"; //This line

// https://astro.build/config
export default defineConfig({
    output: "server", //This line
    adapter: vercel() //This line
});
```

Thanks,
Realluke

**Reply by amyegan** [reply]
Thanks for sharing this. I can use the specific deployment url to see the problem. No need to keep the production version of your site broken any longer! I'm looking into the problem and will update here when I know more

**Reply by amyegan** [reply]
Okay! Updates on what I tried and what I found:
- Tried deploying a copy of the code you shared and observed the same error you saw. Tried removing `sharp` from `package.json` with no affect on results.
- Tried a new project using the Astro basic template and adding the serverless adapter from the [guide](https://docs.astro.build/en/guides/deploy/vercel/) you shared. That deployed and ran without error.
- Then, I started looking into the other packages in the app code you shared to identify differences between the two examples. I removed `import { SocialImage } from "astro-social-images"` and the `SocialImages` component from `Posts.astro` which seems to have solved the problem (at least as a workaround). So I'm thinking there may be some interaction between `astro-social-images` and the `@astrojs/vercel/serverless` adapter that is causing the error.  

I would recommend reaching out to the Astro team at [astro.build/chat](https://astro.build/chat) since they may know more about their adapter and possible interactions that have been reported. I would also reach out in https://github.com/Princesseuh/astro-social-images to let them know about the error you saw since they might also have seen a similar issue and have a solution.

**Reply by reallukedev** [reply]
> Okay! Updates on what I tried and what I found:
> 
> * Tried deploying a copy of the code you shared and observed the same error you saw. Tried removing `sharp` from `package.json` with no affect on results.
> * Tried a new project using the Astro basic template and adding the serverless adapter from the [guide](https://docs.astro.build/en/guides/deploy/vercel/) you shared. That deployed and ran without error.
> * Then, I started looking into the other packages in the app code you shared to identify differences between the two examples. I removed `import { SocialImage } from "astro-social-images"` and the `SocialImages` component from `Posts.astro` which seems to have solved the problem (at least as a workaround). So I'm thinking there may be some interaction between `astro-social-images` and the `@astrojs/vercel/serverless` adapter that is causing the error.
> 
> I would recommend reaching out to the Astro team at [astro.build/chat](https://astro.build/chat) since they may know more about their adapter and possible interactions that have been reported. I would also reach out in https://github.com/Princesseuh/astro-social-images to let them know about the error you saw since they might also have seen a similar issue and have a solution.

Thank you! I will reach out to them!
